functionary/builder/utils.py

```diff
@@ -192,7 +193,45 @@ def _create_package_from_definition(
     package_obj.language = package_definition.get("language")
     package_obj.image_name = image_name
 
-    # TODO: In the same transaction, create the functions for the package
+    _create_functions_from_definition(package_definition.get("functions"), package_obj)
     package_obj.save()
 
     return package_obj
+
+
+def _create_functions_from_definition(definitions, package: Package):
+    """Creates the functions in the package from the definition file"""
+    for function_def in definitions:
+        name = function_def.get("name")
+        try:
+            function_obj = Function.objects.get(package=package, name=name)
+        except Function.DoesNotExist:
+            function_obj = Function(package=package, name=name)
+
+        function_obj.display_name = function_def.get("display_name")
+        function_obj.description = function_def.get("description")
+        function_obj.schema = json.loads(
+            _generate_function_schema(name, function_def.get("parameters"))
+        )
+        function_obj.save()
+
+
+def _generate_function_schema(name: str, parameters) -> str:
+    """Creates a pydantic model from the parameter definitions and returns the schema
+    as a JSON string"""
+    type_map = {"int": int, "str": str}
```

Comment:
We definitely need at least a TODO to flesh out this type map.

Reply:
done